### PR TITLE
Fix coverall in the same way as recent PRs on neo library

### PR DESCRIPTION
**Pull request by vncoelho**
similar to https://github.com/neo-project/neo/pull/3162

### CODE REVIEW

**Comment by vncoelho** [comment]
It looks very good now!

**Comment by vncoelho** [comment]
`coverage/coveralls — Coverage increased (+1.1%) to 32.136% `

**Comment by shargon** [comment]
We need to exclude neo, like in dev-pack

**Comment by vncoelho** [comment]
> We need to exclude neo, like in dev-pack

I do not understand clearly, can you proceed?

**Comment by vncoelho** [comment]
> > We need to exclude neo, like in dev-pack
> 
> I do not understand clearly, can you proceed?

I see now

```yaml
| Module                     | Line   | Branch | Method |
+----------------------------+--------+--------+--------+
| Neo.Extensions             | 42.85% | 0%     | 40%    |
+----------------------------+--------+--------+--------+
| Neo.VM                     | 8.56%  | 3.05%  | 17.6%  |
+----------------------------+--------+--------+--------+
| Neo.Cryptography.BLS12_381 | 0%     | 0%     | 0%     |
+----------------------------+--------+--------+--------+
| RpcClient                  | 86.06% | 56.2%  | 91.42% |
+----------------------------+--------+--------+--------+
| Neo.IO                     | 39.4%  | 29.41% | 40.62% |
+----------------------------+--------+--------+--------+
| Neo.Json                   | 35.7%  | 15.33% | 40.46% |
+----------------------------+--------+--------+--------+
| Neo                        | 23.11% | 10.88% | 20.8%  |
+----------------------------+--------+--------+--------+
```

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -28,15 +28,17 @@
...
+        dotnet test tests/Neo.Plugins.Storage.Tests /p:CollectCoverage=true /p:CoverletOutput=${GITHUB_WORKSPACE}/coverage-join/ /p:MergeWith=${GITHUB_WORKSPACE}/coverage-join/coverage.json -p:Exclude=\"[Neo.SmartContract.TestEngine]*,[Neo.Compiler.CSharp.UnitTests]*,[Neo]*,[Neo.IO]*,[Neo.Json]*,[Neo.VM]*,[Neo.Extensions]*,[Neo.Cryptography.BLS12_381]*\"
+        dotnet test tests/Neo.Cryptography.MPTTrie.Tests /p:CollectCoverage=true /p:CoverletOutput=${GITHUB_WORKSPACE}/coverage-join/ /p:MergeWith=${GITHUB_WORKSPACE}/coverage-join/coverage.json -p:Exclude=\"[Neo.SmartContract.TestEngine]*,[Neo.Compiler.CSharp.UnitTests]*,[Neo]*,[Neo.IO]*,[Neo.Json]*,[Neo.VM]*,[Neo.Extensions]*,[Neo.Cryptography.BLS12_381]*\"
+        dotnet test tests/Neo.Plugins.OracleService.Tests /p:CollectCoverage=true /p:CoverletOutput=${GITHUB_WORKSPACE}/coverage/lcov  /p:MergeWith=${GITHUB_WORKSPACE}/coverage-join/coverage.json /p:CoverletOutputFormat=lcov -p:Exclude=\"[Neo.SmartContract.TestEngine]*,[Neo.Compiler.CSharp.UnitTests]*,[Neo]*,[Neo.IO]*,[Neo.Json]*,[Neo.VM]*,[Neo.Extensions]*,[Neo.Cryptography.BLS12_381]*\"
```
[the remark this answers is not in the document]
Comment (vncoelho) [reply .github/workflows/main.yml:35]: no , you are wrong

it took us time to find the solution

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -28,15 +28,17 @@ jobs:
...
-        dotnet test tests/Neo.Plugins.RpcServer.Tests /p:CollectCoverage=true /p:CoverletOutput=${GITHUB_WORKSPACE}/coverage/lcov /p:MergeWith=${GITHUB_WORKSPACE}/coverage/coverage.json /p:CoverletOutputFormat=lcov
-        dotnet test tests/Neo.Plugins.Storage.Tests /p:CollectCoverage=true /p:CoverletOutput=${GITHUB_WORKSPACE}/coverage/lcov /p:MergeWith=${GITHUB_WORKSPACE}/coverage/coverage.json /p:CoverletOutputFormat=lcov
-        dotnet test tests/Neo.Cryptography.MPTTrie.Tests /p:CollectCoverage=true /p:CoverletOutput=${GITHUB_WORKSPACE}/coverage/lcov /p:MergeWith=${GITHUB_WORKSPACE}/coverage/coverage.json /p:CoverletOutputFormat=lcov
-        dotnet test tests/Neo.Plugins.OracleService.Tests /p:CollectCoverage=true /p:CoverletOutput=${GITHUB_WORKSPACE}/coverage/lcov /p:MergeWith=${GITHUB_WORKSPACE}/coverage/coverage.json /p:CoverletOutputFormat=lcov
+        dotnet test tests/Neo.Network.RPC.Tests /p:CollectCoverage=true /p:CoverletOutput=${GITHUB_WORKSPACE}/coverage-join/ -p:Exclude=\"[Neo.SmartContract.TestEngine]*,[Neo.Compiler.CSharp.UnitTests]*,[Neo]*,[Neo.IO]*,[Neo.Json]*,[Neo.VM]*,[Neo.Extensions]*,[Neo.Cryptography.BLS12_381]*\"
```
[the remark this answers is not in the document]
Comment (vncoelho) [reply .github/workflows/main.yml:31]: you are correct, thanks

**Comment by vncoelho** [comment]
```yaml
+---------------+--------+--------+--------+
| Module        | Line   | Branch | Method |
+---------------+--------+--------+--------+
| OracleService | 12.47% | 2.99%  | 3.12%  |
+---------------+--------+--------+--------+
| MPTTrie       | 96.23% | 88.46% | 98.36% |
+---------------+--------+--------+--------+
| LevelDBStore  | 52.55% | 73.33% | 47.43% |
+---------------+--------+--------+--------+
| RocksDBStore  | 53.19% | 43.75% | 50%    |
+---------------+--------+--------+--------+
| RpcServer     | 0%     | 0%     | 0%     |
+---------------+--------+--------+--------+
| RpcClient     | 86.06% | 56.2%  | 91.42% |
+---------------+--------+--------+--------+

+---------+--------+--------+--------+
|         | Line   | Branch | Method |
+---------+--------+--------+--------+
| Total   | 50.46% | 39.11% | 58.1%  |
+---------+--------+--------+--------+
| Average | 50.08% | 44.12% | 48.38% |
+---------+--------+--------+--------+
```

**Comment by vncoelho** [comment]
`coverage/coveralls — Coverage increased (+17.0%) to 48.035% `

File: .github/workflows/main.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (vncoelho) [reply .github/workflows/main.yml]: Thanks, improved

**Comment by vncoelho** [comment]
@shargon, this is ready as well

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -28,15 +28,18 @@ jobs:
...
-        dotnet test tests/Neo.Network.RPC.Tests /p:CollectCoverage=true /p:CoverletOutput=${GITHUB_WORKSPACE}/coverage/
-        dotnet test tests/Neo.Plugins.RpcServer.Tests /p:CollectCoverage=true /p:CoverletOutput=${GITHUB_WORKSPACE}/coverage/lcov /p:MergeWith=${GITHUB_WORKSPACE}/coverage/coverage.json /p:CoverletOutputFormat=lcov
-        dotnet test tests/Neo.Plugins.Storage.Tests /p:CollectCoverage=true /p:CoverletOutput=${GITHUB_WORKSPACE}/coverage/lcov /p:MergeWith=${GITHUB_WORKSPACE}/coverage/coverage.json /p:CoverletOutputFormat=lcov
-        dotnet test tests/Neo.Cryptography.MPTTrie.Tests /p:CollectCoverage=true /p:CoverletOutput=${GITHUB_WORKSPACE}/coverage/lcov /p:MergeWith=${GITHUB_WORKSPACE}/coverage/coverage.json /p:CoverletOutputFormat=lcov
-        dotnet test tests/Neo.Plugins.OracleService.Tests /p:CollectCoverage=true /p:CoverletOutput=${GITHUB_WORKSPACE}/coverage/lcov /p:MergeWith=${GITHUB_WORKSPACE}/coverage/coverage.json /p:CoverletOutputFormat=lcov
+        EXCLUDED_TESTS=\"[Neo.SmartContract.TestEngine]*,[Neo.Compiler.CSharp.UnitTests]*,[Neo]*,[Neo.IO]*,[Neo.Json]*,[Neo.VM]*,[Neo.Extensions]*,[Neo.Cryptography.BLS12_381]*\"
```
Comment (shargon) [inline .github/workflows/main.yml:31]:
```suggestion
        EXCLUDED_TESTS=\"[Neo]*,[Neo.IO]*,[Neo.Json]*,[Neo.VM]*,[Neo.Extensions]*,[Neo.Cryptography.BLS12_381]*\"
```

**Review by shargon — APPROVED** [review]

**Pull request merged**